### PR TITLE
Update deepbgc to use prefix for all files, add stub, single setup

### PR DESCRIPTION
**Pull request by jfy133**
## PR checklist

Closes #XXX 

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - For modules:
    - [ ] `nf-core modules test <MODULE> --profile docker`
    - [ ] `nf-core modules test <MODULE> --profile singularity`
    - [ ] `nf-core modules test <MODULE> --profile conda`
  - For subworkflows:
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile docker`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile singularity`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile conda`

### CODE REVIEW

**Comment by jfy133** [comment]
@jasmezz @Darcy220606 this is the one the tests weirdly fail for reasons I don't understand

**Comment by Darcy220606** [comment]
> @jasmezz @Darcy220606 this is the one the tests weirdly fail for reasons I don't understand

Some assertions are failing on the github actions side (although) it doesnt show which one :( ) so i guess theres an issue there. Some snaps are missing content when u use 'match' , maybe its an issue with that ?

**Comment by jfy133** [comment]
Ugh unless some of the files are now too large?

**Comment by jfy133** [comment]
Nope, I think it was a misconfogured test assterion

Hopefully all good now!

File: modules/nf-core/deepbgc/pipeline/main.nf [hunk 1]
```diff
@@ -12,17 +12,17 @@ process DEEPBGC_PIPELINE {
...
+    tuple val(meta), path("${prefix}/LOG.txt")                           ,   emit: log
+    tuple val(meta), path("${prefix}/${prefix}.antismash.json")          ,   optional: true, emit: json
+    tuple val(meta), path("${prefix}/${prefix}.bgc.gbk")                 ,   optional: true, emit: bgc_gbk
+    tuple val(meta), path("${prefix}/${prefix}.bgc.tsv")                 ,   optional: true, emit: bgc_tsv
+    tuple val(meta), path("${prefix}/${prefix}.full.gbk")                ,   optional: true, emit: full_gbk
+    tuple val(meta), path("${prefix}/${prefix}.pfam.tsv")                ,   optional: true, emit: pfam_tsv
+    tuple val(meta), path("${prefix}/evaluation/${prefix}.bgc.png")      ,   optional: true, emit: bgc_png
+    tuple val(meta), path("${prefix}/evaluation/${prefix}.pr.png")       ,   optional: true, emit: pr_png
+    tuple val(meta), path("${prefix}/evaluation/${prefix}.roc.png")      ,   optional: true, emit: roc_png
+    tuple val(meta), path("${prefix}/evaluation/${prefix}.score.png")    ,   optional: true, emit: score_png
```
Comment (jasmezz) [inline modules/nf-core/deepbgc/pipeline/main.nf:25]:
```suggestion
    tuple val(meta), path("${prefix}/README.txt")                    ,   optional: true, emit: readme
    tuple val(meta), path("${prefix}/LOG.txt")                       ,   emit: log
    tuple val(meta), path("${prefix}/${prefix}.antismash.json")      ,   optional: true, emit: json
    tuple val(meta), path("${prefix}/${prefix}.bgc.gbk")             ,   optional: true, emit: bgc_gbk
    tuple val(meta), path("${prefix}/${prefix}.bgc.tsv")             ,   optional: true, emit: bgc_tsv
    tuple val(meta), path("${prefix}/${prefix}.full.gbk")            ,   optional: true, emit: full_gbk
    tuple val(meta), path("${prefix}/${prefix}.pfam.tsv")            ,   optional: true, emit: pfam_tsv
    tuple val(meta), path("${prefix}/evaluation/${prefix}.bgc.png")  ,   optional: true, emit: bgc_png
    tuple val(meta), path("${prefix}/evaluation/${prefix}.pr.png")   ,   optional: true, emit: pr_png
    tuple val(meta), path("${prefix}/evaluation/${prefix}.roc.png")  ,   optional: true, emit: roc_png
    tuple val(meta), path("${prefix}/evaluation/${prefix}.score.png"),   optional: true, emit: score_png
```

File: modules/nf-core/deepbgc/pipeline/tests/main.nf.test [hunk 2]
```diff
@@ -108,9 +88,30 @@ nextflow_process {
...
+
+        then {
+            assertAll(
+                { assert process.success },
+                { assert snapshot(process.out).match() }
+            )
+        }
+    }
+
+
```
Comment (jasmezz) [inline modules/nf-core/deepbgc/pipeline/tests/main.nf.test:115]:
```suggestion
```

**Review by jasmezz — APPROVED** [review]
LGTM! :cat:

**Pull request merged**